### PR TITLE
Feature/19

### PR DESCRIPTION
**Pull request by bspitzbart**
Closes issues 10, 12, 16, 19
Tests shown in deploy_sealnet.ipynb
predict_raster_det.py replaces predict_raster.py

### CODE REVIEW

File: predict_raster_det.py [hunk 1]
```diff
@@ -169,51 +175,118 @@ def main():
...
     if len(pos_patches) > 0:
 
-        if not os.path.exists('./{}/sub-tiles/images'.format(args.dest_folder)):
-            os.makedirs('./{}/sub-tiles/images'.format(args.dest_folder))
+        if args.count_architecture is not None:
+            count_size = model_archs[args.count_architecture]['input_size']
+        else:
+            count_size = model_archs[args.det_architecture]['input_size']
+
+        os.makedirs('./{}/subtiles/images'.format(args.dest_folder))
```
Comment (iparask) [inline predict_raster_det.py:183]: Wouldn't it be better to let the user of this script decide the outptut folder structure?

**Review by iparask — CHANGES_REQUESTED** [review]
I am not sure if I am the right person for this PR. It closes issues #10 , #12, #16 and #19.

There is no information in those issues that will allow me to understand the proposed changes. By scanning the files, I can suggest some small changes.

The notebook should not be part of this repo, at least based on our discussion from the last CI meeting. So I would suggest to remove it. 

I would like to avoid any hardcoded paths or folder trees generated from the script, unless it has to do with file names. So, I suggest that any path that any script needs to write should be an input. If it is writing temporary files then do as you think it is best.

I want to suggest that @bentocg does also a review, since he is the expert in what the kernels should be doing.

Also, I do not really understand some the file names.

**Comment by iparask** [comment]
Also, I realized that @bentocg did all the coding and @bspitzbart opened the PR. That creates an issue who should review and who should not. I may not be the best suited person for every review.

File: predict_raster_det.py (hunk 1 above)
Comment (bentocg) [reply predict_raster_det.py:183]: Hi Giannis, 
I let the user decide the root output folder. We can add optional inputs for folder names but the dataloaders require subfolders (I know I could change that, but its easier to create this subfolder). Also, tiles and subtiles all get deleted once the prediction is done so we don't have to worry to much about naming here.

Comment (iparask) [reply predict_raster_det.py:183]: Okay

**Review by iparask — APPROVED** [review]

**Pull request merged**